nixos/default.nix

```diff
@@ -79,7 +82,70 @@ in {
               '';
             in "${setupEnv} ${usercfg.home.activationPackage}";
           };
+        })) cfg.users;
+    })
+    (mkIf (cfg.users != { } && cfg.useUserService) {
+      systemd.user.services.home-manager = baseUnit // {
+        description = "Home Manager environment for %u";
+        wantedBy = [ "default.target" ];
+
+        # user units cannot depend on system units
+        # TODO: Insert in the script logic for waiting on the nix socket via dbus
+        # like https://github.com/mogorman/systemd-lock-handler
+        # wants = [ "nix-daemon.socket" ];
+        # after = [ "nix-daemon.socket" ];
+
+        unitConfig.RequiresMountsFor = "%h";
+        serviceConfig = {
+          Type = "oneshot";
+          RemainAfterExit = "yes";
+          TimeoutStartSec = 90;
+          SyslogIdentifier = "hm-activate-%u";
+          # no ExecStart= is defined for any user that has not defined
+          # config.home-manager.users.${username}
+          # this will be overridden by the below drop-in
+        };
+      };
+
+      users.users = mapAttrs (_: usercfg: {
+        packages =
+          # unit files are taken from $XDG_DATA_DIRS too
+          # but are loaded after units from /etc
+          # we write a drop in so that it will take precedence
+          # over the above unit declaration
+          [
+            (pkgs.writeTextDir "${hmDropIn}/10-user-activation.conf" ''
+              [Service]
+              ExecStart=${usercfg.home.activationPackage}/activate
+            '')
+          ];
+      }) cfg.users;
+      environment.pathsToLink = [ hmDropIn ];
+
+      systemd.services = mapAttrs' (_: usercfg:
+        let inherit (usercfg.home) username homeDirectory;
+        in nameValuePair "ssh_config-${utils.escapeSystemdPath username}" {
+          inherit (usercfg.programs.ssh) enable;
+          description = "Linking ${username}' ssh config";
+          wantedBy = [ "multi-user.target" ];
+          before = [ "systemd-user-sessions.service" ];
+
+          unitConfig.RequiresMountsFor = homeDirectory;
+          stopIfChanged = false;
+
+          serviceConfig = {
+            User = username;
+            Type = "oneshot";
+            RemainAfterExit = "yes";
+            TimeoutStartSec = 90;
+            SyslogIdentifier = "hm-activate-${username}";
+
+            ExecStart = [
+              "${pkgs.coreutils}/bin/ln -s ${usercfg.programs.ssh.configPath} ${homeDirectory}/.ssh/config"
```

Comment:
I'm not able to see a specific point home-manager's ssh module where the user's authorized keys are defined.

Reply:
Ah, it seems like the SSH module doesn't do this because of permission issues, see https://github.com/nix-community/home-manager/pull/9#issuecomment-300053034 and https://github.com/nix-community/home-manager/issues/3090#issuecomment-1368807332.

Maybe we can do something like `users.users.${name}.openssh.authorizedKeys.keyFiles = [ "${config.home-files}/.ssh/authorized_keys" ];`?